CRM/Twingle/Form/Profile.php

```diff
@@ -616,9 +613,6 @@ public static function getLocationTypes() {
    */
   public static function getXCMProfiles() {
     if (!isset(static::$_xcm_profiles)) {
-      static::$_xcm_profiles = array(
-          '' => E::ts("&lt;default profile&gt;"),
-      );
```

Comment:
I am not sure if I get your point correctly, if I am using the `master` branch, there are always at least two options for the XCM profile drop-down:
- `<default profile>`  <- is selected
-  `Profil 'default'`

![Bildschirmfoto vom 2023-08-16 17-39-13](https://github.com/systopia/de.systopia.twingle/assets/24728186/b10bb10b-1235-45c2-9e4f-ba9dcfcd949b)

If I select the `<default profile>` option, I get the error message that the XCM profile field is required on saving.

When we apply the code above, the `Profil 'default'` option remains available in the drop down and it is already selected. I can save the profile without problems.

![Bildschirmfoto vom 2023-08-16 17-37-54](https://github.com/systopia/de.systopia.twingle/assets/24728186/1cbd3146-c90d-43ac-8c49-37754bc90f2f)

If I then add another XCM profile, it appears in the drop-down menu and I can select it.

Reply:
![grafik](https://github.com/systopia/de.systopia.twingle/assets/24728186/f40f2e1b-a4a5-4c7a-9037-4ab3ccca31cc)

Reply:
> When we apply the code above, the Profil 'default' option remains available in the drop down and it is already selected.

That's only because of the order the profiles are listed in that dropdown field, and technically, it's not selected, but only the first option, which dropdown fields fall back to when rendering the form and no "empty option" (with no value) is present. The *< default profile >* option kind of like serves as a reminder that until someone tries to save the form, the default XCM profile would have been used. You get the error message about this field now being mandatory, and also get a hint which profile was used before. Without that "empty option", the first item in the dropdown list would be selected and silently overwrite the until-then unset option with whatever has been the first profile in the list.

We *could* make sure that the default XCM profile will always be the first item in that list for keeping that behavior *and* getting rid of the "empty option". We could also just rename the "empty option" to something like *Select Profile* and add a description to the field, saying that historically, the default XCM profile would have been used.

I'm just trying to avoid scenarios where people have old config with no XCM profile selected and on the next *save* of the Twingle profile they start using another XCM profile which they didn't explicitly select.

Reply:
Okay, I like the renaming to "empty" or "please choose a profile". It was confusing to me that there were two default profile options, only one of which worked - until I took a look at the code.